kafkaesk/retry.py

```diff
@@ -0,0 +1,157 @@
+from .exceptions import UnhandledMessage
+from .metrics import MESSAGE_FAILED
+from .metrics import MESSAGE_REQUEUED
+from abc import ABC
+from aiokafka.structs import ConsumerRecord
+from pydantic import BaseModel
+from typing import Callable
+from typing import TYPE_CHECKING
+
+import datetime
+
+if TYPE_CHECKING:
+    from .app import Application
+    from .app import Subscription
+
+
+class RetryInfo(BaseModel):
+    retry_count: int
+    retry_delay: int = 0
+    retry_timestamp: datetime.datetime
+    publish_timestamp: datetime.datetime
+    publish_topic: str
+    error: str
+
+
+class RetryMessage(BaseModel):
+    retry_info: RetryInfo
+    original_record: ConsumerRecord
+
+    class Config:
+        arbitrary_types_allowed = True
+
+
+class FailureInfo(BaseModel):
+    retry_count: int
+    failure_timestamp: datetime.datetime
+    publish_topic: str
+    error: str
+
+
+class FailureMessage(BaseModel):
+    failure_info: FailureInfo
+    original_record: ConsumerRecord
```

Comment:
How is `ConsumerRecord` serialized? That works with pydantic?

Reply:
Good catch, I thought this was a built-in `dataclasses.dataclass` but it's a `namedtuple`.  I'll have to adjust this a bit.

Reply:
I added a pydantic model `Record` which is essentially just a translation layer for a `ConsumerRecord` and that is then used for serialization purposes.  It feels a little bit heavy so if you have a better recommendation I'm open to it.